client/src/JobSelector.jsx

```diff
@@ -1,57 +1,39 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import PropTypes from "prop-types";
 
 import Category from "./Category";
 import AjaxHelper from "./AjaxHelper";
 
-class JobSelector extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = { active: true, data: [] };
-        this.ajax = AjaxHelper.getInstance();
-    }
-
-    hide() {
-        var newState = this.state;
-        newState.active = false;
-        this.setState(newState);
-    }
-
-    show() {
-        this.setState({ active: true, data: [] });
-        this.componentDidMount();
-    }
-
-    componentDidMount() {
-        this.ajax.getJSON(
-            this.ajax.ingestApiUrl,
-            null,
-            function (data) {
-                this.setState({ active: true, data: data });
-            }.bind(this)
-        );
-    }
-
-    render() {
-        var categories = [];
-        var empty_categories = [];
-        for (var i in this.state.data) {
-            var category = this.state.data[i];
-            var element = <Category key={category.category} data={category} />;
-            if (category.jobs.length > 0) {
-                categories[categories.length] = element;
-            } else {
-                empty_categories[empty_categories.length] = element;
-            }
-        }
-        return (
-            <div className={this.state.active ? "" : "hidden"} id="jobSelector">
-                {categories}
-                {empty_categories}
-            </div>
-        );
-    }
-}
+const JobSelector = () => {
+    const [active, setActive] = useState(true);
```

Comment:
I'm not sure if there's actually a reason to have the active state anymore -- before we used React router, there was logic to switch back and forth between components by hiding and unhiding things... but now that we're using proper routing, none of that is necessary. I think this may be a relic of that time.

Reply:
@demiankatz seems worth removing this and testing just to make sure state persists the same way

Reply:
So we don't want to hide the jobs anymore? This active property sets the className of the job selector.

Reply:
Oh. Wait. Now I see. This doesn't control hiding the category componet.s

Reply:
Okay. I'm going to remove it.

Reply:
Exactly, it just hides the whole component... and since the component is likely empty and invisible until data loads, hiding it until data loads seems redundant.